master/docs/manual/cfg-workers-docker.rst

```diff
@@ -363,3 +363,80 @@ In addition to the arguments available for any :ref:`Latent-Workers`, :class:`Ma
 .. _txrequests: https://pypi.python.org/pypi/txrequests
 .. _treq: https://pypi.python.org/pypi/treq
 .. _requests authentication plugin: http://docs.python-requests.org/en/master/user/authentication/
+
+Kubernetes latent worker
+========================
+
+Kubernetes_ is an open-source system for automating deployment, scaling, and management of containerized applications.
+
+Buildbot supports using Kubernetes_ to host your latent workers.
+
+.. py:class:: buildbot.worker.kubernetes.KubeLatentWorker
+.. py:class:: buildbot.plugins.worker.KubeLatentWorker
+
+The :class:`KubeLatentWorker` attempts to instantiate a fresh image for each build to assure consistency of the environment between builds
+Each image will be discarded once the worker finished processing the build queue (i.e. becomes ``idle``).
+See :ref:`build_wait_timeout <Common-Latent-Workers-Options>` to change this behavior.
+
+.. _Kubernetes: https://kubernetes.io/
+
+In addition to the arguments available for any :ref:`Latent-Workers`, :class:`KubeLatentWorker` will accept the following extra ones:
+
+``password``
+    (mandatory)
+    The worker password part of the :ref:`Latent-Workers` API.
+    If the password in ``None``, then it will be automatically generated from random number.
+
+``job``
+    (optional, default to KubeLatentWorker.default_job())
+    This is the V1JobSpec_ object to the kubernetes API. You will find documentation about python object on the doc of the python kubernetes-client_.
+    If you want to pass renderable in Job definition you need to wrap your object with ``KubeRenderable``.
+    You can get a default job that references the `official buildbot image`_
+
+``namespace``
+    (optional, default to ``default``)
+    This is the name of the namespace.
+
+``masterFQDN``
+    (optional, default depend of ``getMasterMethod``)
+    Address of the master the worker should connect to. Use if you want to place a loadbalancer between the workers and the masters.
+    See the section ``getMasterMethod`` to see the default value.
+
+``getMasterMethod``
+    (optional, default to ``auto_ip``)
+    If ``None`` passed as ``masterFQDN`` or no ``masterFQDN`` passed, determines the default value of ``masterFQDN``:
+
+    * ``auto_ip`` scheme will set ``masterFQDN=socker.gethostbyname(socket.getfqdn())``
+    * ``fqdn`` scheme will set ``masterFQDN=socket.getfqdn()``
+
+    ``auto_ip`` is used by default as it fits better the kubernetes case
+
+``kubeConfig``
+    (optional)
+    Override python kubernetes-client configuration. For example, you can disable ssl verification when use in cluster configuration::
```

Comment:
I think we also need to have an extraKubeSpec, which will be merged with the default job spec built with masterFQDN, etc.
like I did with marathon.